🐛 [Bug] Team Create API 쿼리 잘못 쓴 부분 수정

### Problem

Optional에 결과값이 여러개 들어와서 500에러 생기는 부분 수정

### Reproduction

팀 생성 API 보내기


### Expected Behavior

팀이 생성됨

### Log

```Shell
Caused by: javax.persistence.NonUniqueResultException: query did not return a unique result: 2
```


### Browser

Chrome

--- a/gg-agenda-api/src/main/java/gg/agenda/api/user/agendateam/controller/response/OpenTeamResDto.java
+++ b/gg-agenda-api/src/main/java/gg/agenda/api/user/agendateam/controller/response/OpenTeamResDto.java
@@ -1,6 +1,9 @@
 package gg.agenda.api.user.agendateam.controller.response;
 
+import java.util.List;
+
 import gg.data.agenda.AgendaTeam;
+import gg.data.agenda.type.Coalition;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 
@@ -11,11 +14,13 @@ public class OpenTeamResDto {
 	private String teamLeaderIntraId;
 	private int teamMateCount;
 	private String teamKey;
+	private List<Coalition> coalitions;
 
-	public OpenTeamResDto(AgendaTeam agendaTeam) {
+	public OpenTeamResDto(AgendaTeam agendaTeam, List<Coalition> coalitions) {
 		this.teamName = agendaTeam.getName();
 		this.teamLeaderIntraId = agendaTeam.getLeaderIntraId();
 		this.teamMateCount = agendaTeam.getMateCount();
 		this.teamKey = agendaTeam.getTeamKey().toString();
+		this.coalitions = coalitions;
 	}
 }

--- a/gg-agenda-api/src/main/java/gg/agenda/api/user/agendateam/service/AgendaTeamService.java
+++ b/gg-agenda-api/src/main/java/gg/agenda/api/user/agendateam/service/AgendaTeamService.java
@@ -130,9 +130,10 @@ public class AgendaTeamService {
 			throw new BusinessException(LOCATION_NOT_VALID);
 		}
 
-		agendaTeamProfileRepository.findByAgendaProfileAndIsExistTrue(agenda, agendaProfile).ifPresent(teamProfile -> {
-			throw new DuplicationException(AGENDA_TEAM_FORBIDDEN);
-		});
+		agendaTeamProfileRepository.findByAgendaAndProfileAndIsExistTrue(agenda, agendaProfile)
+			.ifPresent(teamProfile -> {
+				throw new DuplicationException(AGENDA_TEAM_FORBIDDEN);
+			});
 
 		if (agenda.getIsOfficial()) {
 			Ticket ticket = ticketRepository.findByAgendaProfileAndIsApprovedTrueAndIsUsedFalse(agendaProfile)

--- a/gg-agenda-api/src/main/java/gg/agenda/api/user/agendateam/service/AgendaTeamService.java
+++ b/gg-agenda-api/src/main/java/gg/agenda/api/user/agendateam/service/AgendaTeamService.java
@@ -225,11 +225,16 @@ public class AgendaTeamService {
 	public List<OpenTeamResDto> listOpenTeam(UUID agendaKey, Pageable pageable) {
 		Agenda agenda = agendaRepository.findByAgendaKey(agendaKey)
 			.orElseThrow(() -> new NotExistException(AGENDA_NOT_FOUND));
-
 		List<AgendaTeam> agendaTeams = agendaTeamRepository.findByAgendaAndStatusAndIsPrivateFalse(agenda, OPEN,
 			pageable).getContent();
 		return agendaTeams.stream()
-			.map(OpenTeamResDto::new)
+			.map(agendaTeam -> {
+				List<Coalition> coalitions = agendaTeamProfileRepository
+					.findByAgendaTeamAndIsExistTrue(agendaTeam).stream()
+					.map(agendaTeamProfile -> agendaTeamProfile.getProfile().getCoalition())
+					.collect(Collectors.toList());
+				return new OpenTeamResDto(agendaTeam, coalitions);
+			})
 			.collect(Collectors.toList());
 	}
 

--- a/gg-repo/src/main/java/gg/repo/agenda/AgendaTeamProfileRepository.java
+++ b/gg-repo/src/main/java/gg/repo/agenda/AgendaTeamProfileRepository.java
@@ -19,6 +19,8 @@ public interface AgendaTeamProfileRepository extends JpaRepository<AgendaTeamPro
 	@Query("SELECT atp FROM AgendaTeamProfile atp WHERE atp.agendaTeam = :agendaTeam AND atp.isExist = true")
 	List<AgendaTeamProfile> findByAgendaTeamAndIsExistTrue(AgendaTeam agendaTeam);
 
+	@Query("SELECT atp FROM AgendaTeamProfile atp WHERE atp.agenda = :agenda AND atp.profile = :agendaProfile "
+		+ "AND atp.isExist = true")
 	Optional<AgendaTeamProfile> findByAgendaAndProfileAndIsExistTrue(Agenda agenda, AgendaProfile agendaProfile);
 
 	/**

